examples/generic-azure.yaml

```diff
@@ -637,4 +637,41 @@ groups:
 
           ## References
           - Cluster installation in https://docs.microsoft.com/en-us/azure/virtual-machines/workloads/sap/high-availability-guide-suse-pacemaker
-        scored: false
\ No newline at end of file
+        scored: false
+      - id: 1.5.2
+        description: "npt service"
+        audit: |
+          if ! sudo systemctl status ntpd; then echo "result=failed"; fi
+          if sudo ntpq -p | grep "No association ID's returned"; then echo "result=failed"; fi
+          if [[ $(sudo ntpq -p | wc -l) -lt 3 ]]; then echo "result=failed"; fi
```

Comment:
why would it matter if its 3 or more? the important part is one of those is accepted, e.g. there is `grep  '^\*` matching it

Reply:
ah, it just checks whether there is at least one result returned.

Reply:
Yes, correct, the output is something like
```
cloudadmin@vmhana01:~> sudo ntpq -p
     remote           refid      st t when poll reach   delay   offset  jitter
==============================================================================
 20.101.61.75    .INIT.          16 u  597 1024    0    0.000    0.000   0.000

```

Reply:
yeah, this output is actually a fail (stratum is 16, sync is missing). your check makes it a pass

Reply:
> if kvm virtualization is used, assume kvm-clock (e.g. real time clock is passed in from hypervisor host and not emulated within the virtual machine) -> no ntp check, or verify that ntp is not running, because it shouldn't
> in other cases, check that the clock has sync (there is one chosen peer, as indicated by an '*' in the first line.

It's amended. If it's a kvm-guest it checks the kvm-clocks, otherwise, it checks ntp with stratum != 16.

Reply:
@dirkmueller we are already assuming that we are on Azure cloud, which kinda excludes that we're running on a kvm guest, doesn't it? So I would safely assume that we can rely on just ntp.

Reply:
Pretty much every SAP Note related to setting up a system will tell you that it's strongly recommended to set up time sync (i.e. NTP). We also do that in our SLES15 SAP Note. 
I didn't find any indication to deviate from that recommendation for KVM VMs, any other virtualization or cloud. 

When you check the NTP configuration/status, please also make sure it's configured with "slew" option (i.e. only minimal jumps in time when adjusting). Bad things can happen if it's configured with "step".

Reply:
@gereonvey thanks for your feedback!  this was my recollection as well. Ah, yes, the SAP Note is "2578899 - SUSE Linux Enterprise Server 15: Installation Note"

@aleksei-burlakov  can you please remove the KVM exception and also add reference to SAP Note "2578899 - SUSE Linux Enterprise Server 15: Installation Note" in addition to our SUSE HA document.
Can you also check for "slew" as mentioned by Gereon?  (If this is too much for one check, we could consider splitting) @stefanotorresi & @dirkmueller - are you two ok with this?

Reply:
Yes, it's fine: after some googling it seems that running NTP is always a good idea, on any hypervisor, so I wouldn't treat KVM any differently.
@aleksei-burlakov the url of the sap note to refer is: https://launchpad.support.sap.com/#/notes/2578899.

Reply:
Thanks for reviewing, I have

- removed the kvm check
- added the slew check
- changed the reference

Reply:
https://documentation.suse.com/sles/15-SP1/html/SLES-all/sec-kvm-managing-clock.html#sec-kvm-managing-clock-kvm recommends to run ntp even with kvm_clock. so that part seems fine.